src/python/espressomd/constraints.py

```diff
@@ -441,6 +441,60 @@ def phi0(self):
     _so_name = "Constraints::LinearElectricPotential"
 
 
+@script_interface_register
+class ElectricPlaneWave(Constraint):
+
+    """
+    Electric field of the form
+
+      E = E0 * sin(k * x + omega * t + phi)
+
```

Comment:
What is the direction of the electric field? E0 is only the amplitude as described below (i.e. a scalar value). The electric field and the wave vector do not need to coincide if there are charge densities.